lib/trade_tariff_frontend.rb

```diff
@@ -102,4 +102,29 @@ def call(env)
       end
     end
   end
+
+  class FilterBadQueryParameterEncoding
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @query_string = env['QUERY_STRING'].to_s
+      begin
+        Rack::Utils.parse_nested_query @query_string
+      rescue Rack::Utils::InvalidParameterError
+        return bad_request
+      end
+
+      @app.call(env)
+    end
+
+    def bad_request
+      [
+        400,
+        { "Content-Type" => "application/json" },
+        [{ status: 400, error: "There was a problem with your query: '#{@query_string}'" }.to_json]
```

Comment:
@mcumcu for api v2 it should be jsonapi compatible

Reply:
Updated error:
```
{
    "errors": [
        {
            "status": "400",
            "title": "There was a problem with your query",
            "source": {
                "parameter": "query[letter]=%"
            }
        }
    ]
}
```
https://jsonapi.org/format/#errors